packages/oceanfront/src/components/Dialog.vue

```diff
@@ -39,28 +52,172 @@ export default defineComponent({
     id: String,
     loading: Boolean,
     modelValue: Boolean,
+    resize: { type: Boolean, default: false },
+    dragAndDrop: { type: Boolean, default: true },
     transition: { type: String, default: 'slide-down' },
   },
   emits: ['update:modelValue'],
   setup(props, ctx: SetupContext) {
+    const dialog = ref<any>()
+    const dialogHeader: Ref<HTMLDivElement | undefined> = ref()
     const active = ref(props.modelValue)
+
+    const padding = 10
+    const bottomPadding = 85
+
+    let currentZIndex = ref(100)
+
+    let position1 = ref(0)
```

Comment:
I think instead of a bunch of `ref` instances you can use one `reactive({x1, y1, x2, y2}` (for instance) here